### PR TITLE
Support for container local alias

### PR DESCRIPTION
**Pull request by mavenugo**

### CODE REVIEW

File: sandbox.go [hunk 1]
```diff
@@ -424,26 +424,56 @@ func (sb *sandbox) ResolveName(name string) net.IP {
...
 			continue
 		}
 
+		if alias {
+			if ep.aliases == nil {
+				continue
+			}
+
+			var ok bool
+			name, ok = ep.aliases[req]
```
Comment (mrjana) [inline sandbox.go]: Need to acquire endpoint lock while accessing the ep

File: sandbox.go [hunk 2]
```diff
@@ -424,26 +424,56 @@ func (sb *sandbox) ResolveName(name string) net.IP {
...
 	log.Debugf("To resolve %v", parts)
 
+	reqName := parts[0]
+	networkName := ""
+	if len(parts) > 1 {
+		networkName = parts[1]
+	}
+
+	// First check for local container alias
+	ip = sb.resolveName(reqName, networkName, true)
```
Comment (mrjana) [inline sandbox.go]: May be pass the result of sb.getConnectedEndpoints() as a param rather traversing it twice

**Comment by mavenugo** [comment]
@sanimej @mrjana took care of both of your comments.
If a configured alias exists in a network without a backing container and if there is a container with the same name as the alias exists, then the lookup logic will skip that endpoint and try to perform the container name lookup on another network.
PTAL

**Comment by mrjana** [comment]
LGTM

**Pull request merged**